### PR TITLE
Added a new card-flip functionality to cards and framework to add preset datapacks to existing datapacks (Lottie implementation)

### PR DESCRIPTION
**Pull request by paologum**
I added a way to display lotties easily with various features in `TSCLottie.tsx`

Screenshot 2024-02-15 at 4 07 50 PM
tsc-158-output

### CODE REVIEW

File: app/src/Home.tsx [hunk 1]
```diff
@@ -67,138 +77,96 @@ export const Home = observer(function Home() {
...
             actions.removeCache();
             actions.resetState();
-          }}>
-            Remove Cache
+          }}
+        >
+          Remove Cache
         </TSCButton>
       </div>
-      <Drawer anchor="bottom"
```
Comment (paologum) [inline app/src/Home.tsx:74]: no use for drawer now

File: app/src/Home.tsx [hunk 2]
```diff
@@ -67,138 +77,96 @@ export const Home = observer(function Home() {
...
+          >
+            {configArray.map((preset, index) => (
+              <Grid
+                item
+                key={index}
+                style={{ marginRight: "16px", marginLeft: "16px" }}
+              >
+                <TSCCard
+                  color={theme.palette.navbar.main}
+                  preset={preset}
```
Comment (paologum) [inline app/src/Home.tsx:151]: was generalized but now we just take a preset

File: app/src/SettingsTabs/Column.tsx
Comment (paologum) [inline app/src/SettingsTabs/Column.tsx:1]: this file is format only from prettier

File: app/src/SettingsTabs/MapButtons.tsx [hunk 3]
```diff
@@ -25,50 +46,33 @@ export const ChildMapIcon = () => {
   );
 };
 
-export const BorderedIcon = ({
```
Comment (paologum) [inline app/src/SettingsTabs/MapButtons.tsx:28]: decided to move this to better suit the app in general when we want bordered icons

File: app/src/components/TSCLottie.tsx
Comment (paologum) [inline app/src/components/TSCLottie.tsx:1]: this is the most important file for this branch, implements lottie playing functionality given a lottie json

File: app/src/state/state.ts [hunk 4]
```diff
@@ -17,7 +17,6 @@ export type State = {
   chartLoading: boolean;
   tab: number;
   madeChart: boolean;
-  showPresetInfo: boolean;
```
Comment (paologum) [inline app/src/state/state.ts:20]: don't need this anymore

File: shared/src/index.ts [hunk 5]
```diff
@@ -47,10 +47,14 @@ export type ChartConfig = {
   title: string;
   description: string;
   settings: string; // path to base settings file
-  datapacks: string[]; // active datapack names
+  datapacks: Datapack[]; // active datapack names
   date: string; // active datapack names
   type?: string; // type of preset
 };
+export type Datapack = {
```
Comment (paologum) [inline shared/src/index.ts:54]: allows us to have more usable names for display of datapacks

**Review by jay-jhk-lee — APPROVED** [review]
Looks Good

**Pull request merged**